nntrainer/include/layer.h

```diff
@@ -425,6 +427,7 @@ class Layer {
    * @brief     Gradient for the weights in this layer
    */
   std::vector<std::reference_wrapper<Tensor>> gradients;
```

Comment:
How about remove `reference_wrapper` here and use this variable as local variable inside `backwarding`  and lift this to a state saving vector? It seems that it is needed only for applying gradient.

```suggestion
  std::vector<Tensor> gradients;
```

Reply:
Yes, thats good.
But this would need the copy constructor of Tensor to be default and not create another copy of itself.
Lets do this after #258.

Reply:
Yes. I'll do that :)